src/handlers/handleOptionChange/handleOptionChange.tsx

```diff
@@ -0,0 +1,20 @@
+interface CurrentOptionHandler {
+  event: React.ChangeEvent<HTMLSelectElement>;
+  setPerPage: (pageNumber: number) => void;
+  setCurrentPage: (pageNumber: number) => void;
+  newParams: URLSearchParams;
+  setSearchParams: (currentOptions: URLSearchParams) => void;
+}
+
+export default function handleOptionChange(params: CurrentOptionHandler): void {
+  const { event, setPerPage, setCurrentPage, newParams, setSearchParams } =
+    params;
+  const startPage = import.meta.env.VITE_START_PAGE;
+
+  setPerPage(+event.target.value);
```

Comment:
The `setPerPage` function is called with the value from the event target. Ensure that `event.target.value` is always a valid number to avoid potential issues.